evm/src/arithmetic/mod.rs

```diff
@@ -119,13 +119,19 @@ impl TernaryOperator {
     }
 }
 
+/// An enum representing arithmetic operations that can be either binary or ternary.
+///
+/// Binary operations include a special `is_simulated` flag to differentiate SHL
+/// and SHR operations from MUL and DIV respectively, as the former are simulated
+/// with the latter by scaling their inputs.
```

Comment:
I just made a PR https://github.com/topos-protocol/plonky2/pull/3 with a suggestion for a potentially cleaner way to solve this problem. Let me know what you think.